packages/recommend-react/src/useRecommendations.ts

```diff
@@ -19,35 +20,48 @@ export function useRecommendations<TObject>({
   recommendClient,
   threshold,
   transformItems: userTransformItems,
-}: GetRecommendationsProps<TObject>) {
-  const [result, setResult] = useState<GetRecommendationsResult<TObject>>({
-    recommendations: [],
-  });
+  initialState: userInitialState,
+}: GetRecommendationsProps<TObject> & {
+  initialState?: InitialResults<TObject>;
+}) {
+  const isFirstRenderRef = useRef(false);
+
   const { status, setStatus } = useStatus('loading');
   const objectIDs = useStableValue(userObjectIDs);
   const transformItems = useStableValue(userTransformItems);
+  const initialState = useStableValue(userInitialState);
```

Comment:
We shouldn't need the newly introduced variables with something like that:

```tsx
const initialState = useStableValue<GetRecommendationsResult<TObject>>({
  recommendations: [],
  ...userInitialState,
});
const [result, setResult] = useState<GetRecommendationsResult<TObject>>(
  initialState
);
```

Reply:
✅